services/ui_backend_service/api/task.py

```diff
@@ -28,7 +28,7 @@ def __init__(self, app, db, cache=None):
             self.get_task_attempts,
         )
         self._async_table = self.db.task_table_postgres
-        self.refiner = TaskRefiner(cache=cache)
+        self.refiner = TaskRefiner(cache=cache.artifact_cache if cache else None)
```

Comment:
should the else None be outside the init perhaps? as is the case in `ws.py`

Reply:
Good point